### PR TITLE
OncoPrint figures

### PR DESCRIPTION
**Pull request by jaclyn-taroni**
⚠️  Stacked on #1193 

Addressing #1177 and #1178, but made slightly different choices than what is detailed in those issues. Importantly, I've only removed cancer groups with less than 10 samples for the other CNS plot. In the current version of #1193, cancer groups with n < 10 in the other 3 broad histologies are randomly assigned a hex code from the `Greys` color brewer palette. There are over 20 cancer groups with n < 10 that'd be included in the Other CNS plot (fewer than that when you consider which groups actually show up in the plots using the other CNS genes of interest list), so it's not feasible to randomly assign greys to them in the same way. Instead, the Other CNS cancer groups with n < 10 are in a supplemental PDF where they are assigned colors based on the `Paired` color brewer palette.

Similar to #1194, I've created an Rscript (`figures/scripts/fig2-oncoprint-landscape.R`) that creates individual panels & legends as PDFs (`figures/pdfs/fig2/panels`) and then I've assembled a draft figure in Illustrator. You can see that draft as `figures/pngs/figure2.png`. I've removed the old script that used `multipanelfigure` to assemble the figure from the PNGs in the module (`figures/scripts/oncoprint-landscape.R`).

I needed to rerun the `oncoprint-landscape` module because we had been manually updating the OncoPrint palette, rather than updating the script that has all the palettes, so some colors got dropped in #1161 when the tumor descriptor palette was created and those colors were missing when things were run as part of #1189. I've also deprecated the former figure 2 script to hopefully avoid confusion.

### CODE REVIEW

**Comment by jaclyn-taroni** [comment]
This is a large number of changed files, but everything in `analyses/oncoprint-landscape/plots` is just a rerun now that I've fixed the oncoprint palette and I'm removing a good number of figures that are deprecated.

File: figures/palettes/oncoprint_color_palette.tsv [hunk 1]
```diff
@@ -11,11 +11,13 @@ In_Frame_Ins	#FFE4E1
 Stop_Codon_Ins	#CC79A7
 Start_Codon_Del	#56B4E9
 Fusion	#7B68EE
-Multi_Hit	#00F021
+Multi_Hit	#CCCCCC
 Hom_Deletion	#313695
 Hem_Deletion	#abd9e9
```
Comment (jharenza) [inline figures/palettes/oncoprint_color_palette.tsv:16]:
```suggestion
```
We are not actually using these

File: figures/palettes/oncoprint_color_palette.tsv [hunk 2]
```diff
@@ -11,11 +11,13 @@ In_Frame_Ins	#FFE4E1
...
-amplification	#c51b7d
-loss	#0072B2
-gain	#D55E00
+Del	#0072B2
+Amp	#D55E00
```
Comment (jharenza) [inline figures/palettes/oncoprint_color_palette.tsv:18]:
```suggestion
Del	#386db0
Amp	#c51b7d
```
new color suggestions

File: figures/palettes/oncoprint_color_palette.tsv [hunk 3]
```diff
@@ -11,11 +11,13 @@ In_Frame_Ins	#FFE4E1
...
 High_Level_Gain	#FF0000
 Multi_Hit_Fusion	#CD96CD
+Intron	#5ea4fe
+5'Flank	#386db0
+3'Flank	#c51b7d
```
Comment (jharenza) [inline figures/palettes/oncoprint_color_palette.tsv:23]:
```suggestion
5'Flank	#7CCBAC
3'Flank	#FD8344
```
new color suggestions - moving these two to Amp/Del, where I had meant them to be previously, but it looks like things got shuffled. Let's see what these look like 😬

File: figures/palettes/oncoprint_color_palette.tsv [hunk 4]
```diff
@@ -11,11 +11,13 @@ In_Frame_Ins	#FFE4E1
...
-Multi_Hit	#00F021
+Multi_Hit	#CCCCCC
 Hom_Deletion	#313695
 Hem_Deletion	#abd9e9
-amplification	#c51b7d
-loss	#0072B2
-gain	#D55E00
+Del	#0072B2
+Amp	#D55E00
 High_Level_Gain	#FF0000
```
Comment (jharenza) [inline figures/palettes/oncoprint_color_palette.tsv:19]:
```suggestion
```
We are not using this one either

**Review by jharenza — COMMENTED** [review]
>Instead, the Other CNS cancer groups with n < 10 are in a supplemental PDF where they are assigned colors based on the Paired color brewer palette.

I like this! I will modify Supplemental Figure 3 in the manuscript repo to add this panel.

For the `oncoprint_alteration_legend.pdf`, it looks like we have a few alterations/colors which we do not use due to the renaming of CNV alterations, so I updated the code to remove those. I also made a few suggestions for swapping some colors.

The only other minor thing I saw was that the oncoprint panel pdfs do not contain `goi` in the filename while the analysis module does, so maybe you can add this to be consistent (I double checked you are using goi here and you are 😄 ).

File: figures/palettes/oncoprint_color_palette.tsv (hunk 1 above)
Comment (jaclyn-taroni) [reply figures/palettes/oncoprint_color_palette.tsv:16]: We need to edit the script that generates them (`figures/scripts/color_palettes.R`), not this file directly. Editing this file directly caused problems (see the mentions of #1161 and #1189 in the original PR comment).

Comment (jharenza) [reply figures/palettes/oncoprint_color_palette.tsv:16]: gotcha!

**Comment by jaclyn-taroni** [comment]
@jharenza I've updated the color palette and rerun the figures. Can you take a look before I invest any more effort, i.e., attempt to put together the multipanel version?

**Comment by jharenza** [comment]
> @jharenza I've updated the color palette and rerun the figures. Can you take a look before I invest any more effort, i.e., attempt to put together the multipanel version?

The colors I mentioned look fine, but what is "Complex_event"? I hadn't seen that before and it is black, along with nonsense mutations, so that would need a new color if we use it. I would think it would fit into other categories, unless it is an SV of some sort?

**Comment by jaclyn-taroni** [comment]
> but what is "Complex_event"? I hadn't seen that before and it is black, along with nonsense mutations, so that would need a new color if we use it. I would think it would fit into other categories, unless it is an SV of some sort?

A couple notes:

- Complex event is not new as of my most recent update - we can see it here in https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/1200/commits/f6bcf253ac3ceb242e2557e82ad259ea3eb0f729: https://github.com/jaclyn-taroni/OpenPBTA-analysis/blob/f6bcf253ac3ceb242e2557e82ad259ea3eb0f729/analyses/oncoprint-landscape/plots/primary-plus_embryonal_oncoprint.png
- Complex event isn't documented as far as I can tell outside of: https://github.com/PoisonAlien/maftools/blob/db66c013049e04999eca8e6aaa440cfe1eb08703/R/oncoplot.R#L644 which I take to mean that we expect to see it when a gene is impacted by both by a mutation and a copy number alteration of some kind
- It's the same navy (or at least close enough) as Translation Start Site. I'll add it to the palette with another color. I'll probably start with a darker green.

**Comment by jaclyn-taroni** [comment]
@jharenza ready for another look

**Comment by jharenza** [comment]
> Complex event isn't documented as far as I can tell outside of: https://github.com/PoisonAlien/maftools/blob/db66c013049e04999eca8e6aaa440cfe1eb08703/R/oncoplot.R#L644 which I take to mean that we expect to see it when a gene is impacted by both by a mutation and a copy number alteration of some kind

Interesting, I had not noticed that before and I wonder if that was earlier code before the CNVs were added as smaller rectangles in which we are able to see a mutation/fusion + a CNV with two different colors. That begs the question why is it showing as complex rather than the two colors for mut/cnv, unless perhaps a multi-hit SNV + CNV? I can't look into that right now, but probably can later tonight. If you get to it before me, let me know.

**Comment by jaclyn-taroni** [comment]
From a cursory look at the code, I think it might arise when there are multiple CNVs. If we set `writeMatrix = TRUE` when we call `oncoplot()`, that might help confirm/reject that idea.

**Comment by jharenza** [comment]
>From a cursory look at the code, I think it might arise when there are multiple CNVs. If we set writeMatrix = TRUE when we call oncoplot(), that might help confirm/reject that idea.

Just noting that we should explain this in the figure legend.

I can't distinguish between the green for complex event and missense, but otherwise, the flank/amp colors are ok for me.

**Comment by jaclyn-taroni** [comment]
Going to request a review from @cbethell! The main file to take a look at is: `figures/scripts/fig2-oncoprint-landscape.R`. We can add the compiled figures later.

File: figures/generate-figures.sh [hunk 5]
```diff
@@ -77,14 +82,8 @@ bash ${analyses_dir}/oncoprint-landscape/run-oncoprint.sh
...
-for filename in "${filenames[@]}"; do
-
-  ## Run the `oncoprint-landscape` figure assembly script
-  Rscript --vanilla scripts/oncoprint-landscape.R \
-    --lead_filename ${filename} \
-    --png_name pngs/${filename}_oncoprint_landscape.png
-
-done
+# Create single panel PDFs and legends
+Rscript --vanilla figures/figures/scripts/fig2-oncoprint-landscape.R
```
Comment (cbethell) [inline figures/generate-figures.sh:86]:
```suggestion
Rscript --vanilla scripts/fig2-oncoprint-landscape.R
```

**Review by cbethell — APPROVED** [review]
This LGTM! Just left a comment re: fixing the file path to the figure script in `generate-figures.sh` (tested it to make sure it works successfully and it does) 👍

File: figures/generate-figures.sh (hunk 5 above)
Comment (jaclyn-taroni) [reply figures/generate-figures.sh:86]: Oops! That's super weird, thanks for catching it!

**Pull request merged**